lib/Jit/LLILCJit.cpp

```diff
@@ -394,3 +395,107 @@ void __cdecl LLILCJit::fatal(int Errnum, ...) {
 void LLILCJit::signalHandler(void *Cookie) {
   // do nothing
 }
+
+template <typename ObjSetT, typename LoadResult>
+void ObjectLoadListener::operator()(llvm::orc::ObjectLinkingLayerBase::ObjSetHandleT ObjHandles,
+                const ObjSetT &Objs, const LoadResult &LoadedObjInfos) {
+  int I = 0;
+  for (const std::unique_ptr<object::ObjectFile> &PObj : Objs) {
+    const object::ObjectFile &Obj = *PObj;
+    const RuntimeDyld::LoadedObjectInfo &L = *LoadedObjInfos[I];
+
+    OwningBinary<ObjectFile> DebugObjOwner = L.getObjectForDebug(Obj);
+    const ObjectFile &DebugObj = *DebugObjOwner.getBinary();
+
+    // Get the address of the object image for use as a unique identifier
+    const void *ObjData = DebugObj.getData().data();
+    DIContext *DwarfContext = new DWARFContextInMemory(DebugObj);
+
+    // Use symbol info to iterate functions in the object.
```

Comment:
Any idea how this will need to change when we have funclets running around? If not, please leave a TODO and open an issue for debug with EH.